### PR TITLE
feat: adds optional confirmation for power device toggle

### PR DESCRIPTION
**Pull request by pedrolamas**
Implements #386, by adding a new setting that allow to indicate if a confirmation dialog should be shown when the user toggles a device power.

Signed-off-by: Pedro Lamas <pedrolamas@gmail.com>

### CODE REVIEW

**Review by dweight — APPROVED** [review]
LGTM :)

File: src/components/common/SystemCommands.vue [hunk 1]
```diff
@@ -161,9 +161,19 @@ export default class SystemCommands extends Mixins(StateMixin, ServicesMixin) {
...
-    SocketActions.machineDevicePowerToggle(device.device, state, wait)
+  async togglePowerDevice (device: Device, wait?: string) {
+    const confirmOnPowerDeviceChange = this.$store.state.config.uiSettings.general.confirmOnPowerDeviceChange
+    let res: boolean | undefined = true
+    if (confirmOnPowerDeviceChange) {
+      res = await this.$confirm(
+        this.$tc('app.general.simple_form.msg.confirm_power_device_toggle'),
+        { title: this.$tc('app.general.label.confirm'), color: 'card-heading', icon: '$error' }
+      )
+    }
```
Comment (ForsakenHarmony) [inline src/components/common/SystemCommands.vue:172]: Really not a fan of this tbh, maybe something like

```suggestion
    if (
      confirmOnPowerDeviceChange
      && !await this.$confirm(
        this.$tc('app.general.simple_form.msg.confirm_power_device_toggle'),
        {
          title: this.$tc('app.general.label.confirm'),
          color: 'card-heading',
          icon: '$error'
      	}
      )
    ) {
      return
    }
```

Comment (pedrolamas) [reply src/components/common/SystemCommands.vue:172]: I personally prefer writing as you did above, but the reason I wrote it as I did was for consistency purposes, as there was already code written that way!

Comment (eduncan911) [reply src/components/common/SystemCommands.vue:172]: i agree with you both.  however, let's keep it consistent with existing coding techniques for now...

**Review by eduncan911 — APPROVED** [review]
LGTM

**Pull request merged**